src/boot/efi/meson.build

```diff
@@ -102,6 +102,46 @@ if have_gnu_efi
         efi_conf.set10('ENABLE_TPM', get_option('tpm'))
         efi_conf.set('SD_TPM_PCR', get_option('tpm-pcrindex'))
 
+        if get_option('sbat_distro') != ''
+                efi_conf.set_quoted('SBAT_PROJECT', meson.project_name())
+                efi_conf.set_quoted('PROJECT_VERSION', substs.get('PROJECT_VERSION'))
+                efi_conf.set_quoted('PROJECT_URL', substs.get('PROJECT_URL'))
+                efi_conf.set_quoted('SBAT_DISTRO_GENERATION', '@0@'.format(get_option('sbat_distro_generation')))
+                sbatvars = [['sbat_distro', 'ID'],
+                            ['sbat_distro_summary', 'NAME'],
+                            ['sbat_distro_url', 'BUG_REPORT_URL']]
+                foreach sbatvar : sbatvars
+                        value = get_option(sbatvar[0])
+                        if value == '' or value == 'auto'
+                                value = run_command('sh', '-c', '. /etc/os-release && echo $' + sbatvar[1]).stdout().strip()
+                        endif
+                        if value != ''
+                                efi_conf.set_quoted(sbatvar[0].to_upper(), value)
+                        endif
+                endforeach
+
+                sbatvars = [['sbat_distro_pkgname', meson.project_name()],
+                            ['sbat_distro_version', get_option('version-tag')]]
+                foreach sbatvar : sbatvars
+                       value = get_option(sbatvar[0])
+                       if value == ''
+                               value = sbatvar[1]
+                       endif
+                       efi_conf.set_quoted(sbatvar[0].to_upper(), value)
+                endforeach
+                if efi_conf.get('SBAT_DISTRO_VERSION') == '""'
```

Comment:
Also here, why the quotes?

Reply:
Because the previous command of set_quoted, may have set it to an empty string. which when quoted is `'""'`